packages/react/src/table/TableScroller.tsx

```diff
@@ -0,0 +1,333 @@
+import {
+	type ReactNode,
+	useCallback,
+	useEffect,
+	useRef,
+	useState,
+} from 'react';
+import { Box } from '../box';
+import { boxPalette } from '../core';
+import { Flex } from '../flex';
+import { ScrollbarArrowLeftIcon, ScrollbarArrowRightIcon } from '../icon';
+import { Stack } from '../stack';
+
+export type TableScrollerProps = { children: ReactNode };
+
+export function TableScroller({ children }: TableScrollerProps) {
+	const trackRef = useRef<HTMLDivElement>(null);
+	const thumbRef = useRef<HTMLButtonElement>(null);
+	const scrollerRef = useRef<HTMLDivElement>(null);
+	const mousePos = useRef({ x: 0, y: 0 });
+
+	const [isDraggingThumb, setIsDraggingThumb] = useState(false);
+	const [thumbPosition, setThumbPosition] = useState(0);
+	const [thumbWidthRatio, setThumbWidthRatio] = useState(0);
+	const [buttonIntervalId, setButtonIntervalId] = useState<number | null>(null);
+
+	const repositionThumb = useCallback(() => {
+		if (!scrollerRef?.current) {
+			return;
+		}
+
+		setThumbPosition(scrollerRef.current.scrollLeft * thumbWidthRatio);
+	}, [scrollerRef, thumbWidthRatio]);
+
+	const calculateThumbWidth = useCallback(() => {
+		if (!scrollerRef?.current) {
+			return;
+		}
+
+		setThumbWidthRatio(
+			(scrollerRef?.current.offsetWidth - 56) /
+				(scrollerRef?.current.scrollWidth - 56)
+		);
+	}, [scrollerRef, setThumbWidthRatio]);
+
+	useEffect(() => {
+		calculateThumbWidth();
+	}, [calculateThumbWidth]);
+
+	useEffect(() => {
+		if (
+			typeof window === 'undefined' ||
+			!('ResizeObserver' in window) ||
+			!scrollerRef.current
+		) {
+			return;
+		}
+
+		const observer = new ResizeObserver(() => {
+			calculateThumbWidth();
+			repositionThumb();
+		});
+
+		observer.observe(scrollerRef.current);
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [calculateThumbWidth, repositionThumb, scrollerRef]);
+
+	const handleThumbPress = (event: React.MouseEvent | React.TouchEvent) => {
+		event.preventDefault();
+
+		if (event.type === 'mousedown' && 'button' in event && event.button === 0) {
+			setIsDraggingThumb(true);
+			mousePos.current = { x: event.pageX, y: event.pageY };
+		} else if (event.type === 'touchstart' && 'touches' in event) {
+			setIsDraggingThumb(true);
```

Comment:
Is it safe to move this duplicated line (lines 75 and 78) out of the if branches and onto line 73? I see there is a chance that neither of these branches run but I can't think of another `event.type` that could run.

Reply:
Yep, you're right